src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp

```diff
@@ -55,8 +55,21 @@ SmokeCoAlarmServer & SmokeCoAlarmServer::Instance()
 
 bool SmokeCoAlarmServer::SetExpressedState(EndpointId endpointId, ExpressedStateEnum newExpressedState)
 {
+    expressedStateMask |= (1 << to_underlying(newExpressedState));
+
     ExpressedStateEnum expressedState;
     bool success = GetAttribute(endpointId, Attributes::ExpressedState::Id, Attributes::ExpressedState::Get, expressedState);
+    expressedStateMask |= (1 << to_underlying(expressedState));
+
+    // Get the highest priority alarm
+    for (ExpressedStateEnum i : expressedStatePriority)
```

Comment:
This logic needs to be in the app, not in the cluster.  Why are these changes being made?